delivery_carrier_pricelist/models/delivery_carrier.py

```diff
@@ -36,12 +35,14 @@ class DeliveryCarrier(models.Model):
     def rate_shipment(self, order):
         if self.invoice_policy == "pricelist":
             # Force computation from pricelist based on invoicing policy
-            # using a temporary record.
             # Required since core's `rate_shipment` relies on `self.delivery_type`
             # to lookup for the right handler.
-            tmp_rec = self.new(self.copy_data()[0])
-            tmp_rec.delivery_type = "pricelist"
-            result = super(DeliveryCarrier, tmp_rec).rate_shipment(order)
+            # using a 'temp record' with new() won't work since it has NewId
+            # and rate_shipment() compares those
+            tmp_type = self.delivery_type
```

Comment:
Mmmmh, I would say this should be protected.

@AnizR 

```python
    with self.env.norecompute():
        (....)
```

Reply:
You wan't to avoid computation of fileds that's right?

I think that `norecompute` is depreceated.
It should be like this, isn't it? 
```python
with self.env.protecting(<all_fields of self>):
    ...
```